nodes.py

```diff
@@ -264,6 +264,52 @@ def set_range(self, conditioning, start, end):
                                                                 "end_percent": end})
         return (c, )
 
+class ConditioningTimestepInterpolate:
+    @classmethod
+    def INPUT_TYPES(s):
+        return {"required": {"conditioning_1": ("CONDITIONING", ),
+                             "conditioning_2": ("CONDITIONING", ),
+                             "interval_range": ("FLOAT", {"default": 0.25, "min": 0.001, "max": 0.5, "step": 0.001})
+                             }}
+    RETURN_TYPES = ("CONDITIONING",)
+    FUNCTION = "interpolate_conds"
+
+    CATEGORY = "advanced/conditioning"
+
+
+    def interpolate_conds(self, conditioning_1, conditioning_2, interval_range):
+        
+        num_intervals = int(1 / interval_range)
+
+        conditioning_1_intervals = []
+        conditioning_2_intervals = []
+
+        for i in range(num_intervals):
+            start = i * interval_range
+            end = (i + 1) * interval_range
+
+            # Ensure the last interval ends exactly at 1.0
+            if i == num_intervals - 1:
+                end = 1.0
+
+            if i % 2 == 0:
+                conditioning_1_intervals.append(node_helpers.conditioning_set_values(conditioning_1, {"start_percent": start, "end_percent": end}))
+            else:
+                conditioning_2_intervals.append(node_helpers.conditioning_set_values(conditioning_2, {"start_percent": start, "end_percent": end}))
+
+        # Combine intervals for conditioning_1 and conditioning_2
+        conditioning_1_intervals_combined = conditioning_1_intervals[0]
+        for i in range(1, len(conditioning_1_intervals)):
+            conditioning_1_intervals_combined = (conditioning_1_intervals_combined + conditioning_1_intervals[i])
```

Comment:
what's going on here? this seems redundant

Reply:
The iteration through all the intervals or the combine part?

Reply:
the combine part - like you already have a loop at the top to build a list, why are you unbuilding and rebuilding the same list?

Reply:
I didn't at first but conditioning did not accept a list. I am only doing the same operation as the ConfitioningCombine is doing.